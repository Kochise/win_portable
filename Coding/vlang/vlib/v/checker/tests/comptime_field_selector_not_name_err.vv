struct Foo {
	test int
	name string
}

fn test<T>() {
	mut t := T{}
	$for f in T.fields {
		$if f.typ is string {
			t.$f = '3'
		}
	}
}

fn main() {
	test<Foo>()
}
