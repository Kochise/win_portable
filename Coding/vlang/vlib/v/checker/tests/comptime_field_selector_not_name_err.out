vlib/v/checker/tests/comptime_field_selector_not_name_err.vv:10:7: error: expected `string` instead of `FieldData` (e.g. `field.name`)
    8 |     $for f in T.fields {
    9 |         $if f.typ is string {
   10 |             t.$f = '3'
      |                ^
   11 |         }
   12 |     }